code/modules/world_api/helpers.dm

```diff
@@ -0,0 +1,80 @@
+//Init the API at startup
+/hook/startup/proc/setup_api()
+	for (var/path in typesof(/datum/topic_command) - /datum/topic_command)
+		var/datum/topic_command/A = new path()
+		if(A != null)
+			topic_commands[A.name] = A
+			topic_commands_names.Add(A.name)
+		listclearnulls(topic_commands)
+		listclearnulls(topic_commands_names)
+
+	if (config.api_rate_limit_whitelist.len)
+		// To make the api_rate_limit_whitelist[addr] grabs actually work.
+		for (var/addr in config.api_rate_limit_whitelist)
+			config.api_rate_limit_whitelist[addr] = 1
+
+	return 1
+
+/world/proc/api_do_auth_check(var/addr, var/auth, var/datum/topic_command/command)
+	//Check if command is on nothrottle list
+	if(command.no_throttle == 1)
+		log_debug("API: Throttling bypassed - Command [command.name] set to no_throttle")
+	else
+		if(world_api_rate_limit[addr] != null && config.api_rate_limit_whitelist[addr] == null) //Check if the ip is in the rate limiting list and not in the whitelist
+			if(abs(world_api_rate_limit[addr] - world.time) < config.api_rate_limit) //Check the last request time of the ip
```

Comment:
Should implement rate limiting fix, if there is no players world.time stops.

Reply:
Righto. Forgot about that.